scipy/signal/spectral.py

```diff
@@ -511,11 +511,481 @@ def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
     if noverlap is None:
         noverlap = nperseg // 8
 
-    freqs, time, Pxy = _spectral_helper(x, x, fs, window, nperseg, noverlap,
+    freqs, time, Sxx = _spectral_helper(x, x, fs, window, nperseg, noverlap,
                                         nfft, detrend, return_onesided, scaling,
-                                        axis, mode=mode)
+                                        axis, mode='psd')
+
+    return freqs, time, Sxx
+
+
+def check_COLA(window, nperseg, noverlap, tol=1e-10):
+    """
+    Check whether the Constant OverLap Add (COLA) constraint is met
+
+    Parameters
+    ----------
+    window : str or tuple or array_like
+        Desired window to use. See `get_window` for a list of windows and
+        required parameters. If `window` is array_like it will be used
+        directly as the window and its length must match nperseg.
+    nperseg : int
+        Length of each segment.
+    noverlap : int
+        Number of points to overlap between segments.
+    tol : float, optional
+        The allowed variance of a bin's weighted sum from the median bin
+        sum.
+
+    Returns
+    -------
+    verdict : bool
+        `True` if chosen combination satisfies COLA within `tol`, `False`
+        otherwise
+
+    See Also
+    --------
+    stft: Short Time Fourier Transform
+    istft: Inverse Short Time Fourier Transform
+
+    Notes
+    -----
+    In order to enable inversion of an STFT via the inverse STFT in
+    `istft`, the signal windowing must obey the constraint of "Constant
+    OverLap Add" (COLA). This ensures that every point in the input data is
+    equally weighted, thereby avoiding aliasing and allowing full
+    reconstruction.
+
+    Some examples of windows that satisfy COLA:
+        - Rectangular window at 0% & 50% overlap
+        - Bartlett window at 50% overlap
```

Comment:
Can you actually list some of the overlaps with ellipses to indicate that there are more?  Something like

        - Bartlett window at 50%, 75%, 83.3%, ...

or 

        - Bartlett window at 1/2, 3/4, 5/6, 7/8, ...

I didn't realize these were possible until a few days ago, and had been trying to implement the arbitrary-overlap windows of http://dx.doi.org/10.1109/ICASSP.2012.6287885 when I could have just used `hann()` with 7/8ths overlap etc.  (Though adding the arbitrary-overlap function is still a good idea, but I'm confused about the boundary conditions, https://github.com/scipy/scipy/issues/2190, etc.)

https://gist.github.com/endolith/c5b39ab78f1910e99cadaced9b839fc1